loop/notifications.js

```diff
@@ -0,0 +1,39 @@
+var Notifications = function(pubsub) {
+  this.observers = {};
+  this.pubsub = pubsub;
+  this.setupObservers();
+}
+
+Notifications.prototype = {
+  /**
+   * Setup the pub/sub on redis expire keys.
+   *
+   * On each expire notification, trigger the appropriate observers.
+   **/
+  setupObservers: function() {
+    var self = this;
+
+    self.pubsub.on("pmessage", function(pattern, channel, key) {
+      Object.keys(self.observers).forEach(function(prefix) {
+        if (key.indexOf(prefix) === 0) {
+          self.observers[prefix](key);
+        }
+      })
+    });
+
+    self.pubsub.psubscribe("__keyevent*__:expired");
```

Comment:
Could we add the key pattern inside the psubscribe? i.e: ``self.pubsub.psubscribe("__keyevent__roomparticipant.*__:expired");``